### DISCUSSION TITLE
期望 el-table 支持 table-layout：auto

### DISCUSSION THREAD

**Discussion by cnguu** [element-plus/element-plus#5539] [Ideas]
大数据表格展示场景下，需要由内容撑开宽度，单行不换行

希望增加设置 table-layout 的 props，尽管功能有限，但也是需要的

**Comment by msidolphin** [comment]
show-overflow-tooltip 是否满足你的要求呢?

**Reply by cnguu** [reply]
并不能，内容撑开宽度的原因是需要直接展示数据

**Comment by msidolphin** [comment]
即使table-layout为auto，内容仍然可能换行，所以你要的可能只是修改cell的样式，改为不换行之后冻结列会出现问题（左右仅一个没问题）
![image](https://user-images.githubusercontent.com/26672484/152945931-a0223908-35de-427a-bdc7-481ec6301322.png)

**Comment by cnguu** [comment]
@msidolphin 换行不是主要的，主要是内容要把宽度撑开，所以需要table-layout设为auto，ant-design很好做到了这点，尽管功能有限（例如冻结列），但也是需要的

**Reply by msidolphin** [reply]
> @msidolphin 换行不是主要的，主要是内容要把宽度撑开，所以需要table-layout设为auto，ant-design很好做到了这点，尽管功能有限（例如冻结列），但也是需要的

我的意思是即使加上了table-layout: auto，内容还是可能会换行的，换行的解决方法就是把上图中的样式改为nowrap和keep-all，实现宽度自动撑开不给column加宽度就可以了，或者你提供一个demo吧，实在不清楚想要什么样的效果

**Reply by cnguu** [reply]
这个我明白，我会根据使用场景进行二次的封装，现在缺少的是tableLayout的参数，希望可以加上

**Reply by msidolphin** [reply]
> 这个我明白，我会根据使用场景进行二次的封装，现在缺少的是tableLayout的参数，希望可以加上

![image](https://user-images.githubusercontent.com/26672484/152951762-eef9cf98-c1ed-43f8-ac8d-ddae07433c77.png)

这个参数并不是说加就加的，目前的实现需要做比较大的调整，我这边不太清楚具体要实现什么效果，换行的原因我前面解释了，并不是这个table-layout的问题

**Reply by cnguu** [reply]
使用场景是直接完整地展示数据，不设置列宽，由内容撑开，所以需要auto布局，我会修改css为不换行，如果ElTable加参数修改也可以

**Reply by cnguu** [reply]
![image](https://user-images.githubusercontent.com/10758182/152953323-2c29b8db-02c7-4741-849f-20bbc36304a4.png)

**Comment by msidolphin** [comment]
https://user-images.githubusercontent.com/26672484/152956001-3601eef3-3cd8-433f-9da5-113ba8bda02a.mp4

是否想实现这种效果？

```vue
<template>
  <el-radio-group v-model="layout">
    <el-radio-button label="auto">auto</el-radio-button>
    <el-radio-button label="fixed">fixed</el-radio-button>
  </el-radio-group>
  <el-table border :data="tableData" style="width: 100%" :table-layout="layout">
    <el-table-column prop="date" label="Date" />
    <el-table-column prop="name" label="Name" />
    <el-table-column prop="address" label="Address" />
  </el-table>
</template>

<script lang="ts" setup>
import { ref } from 'vue'
const layout = ref('auto')
const tableData = [
  {
    date: '2016-05-03',
    name: 'Tom',
    address: 'No. 189, Grove St, Los Angeles',
  },
  {
    date: '2016-05-02',
    name: 'Tom',
    address: 'No. 189, Grove St, Los Angeles',
  },
  {
    date: '2016-05-04',
    name: 'Tom',
    address: 'No. 189, Grove St, Los Angeles',
  },
  {
    date: '2016-05-01',
    name: 'Tom',
    address: 'No. 189, Grove St, Los Angeles',
  },
]
</script>

```

**Reply by cnguu** [reply]
是的

**Comment by msidolphin** [comment]
https://preview-5860-element-plus.surge.sh/en-US/component/table.html#table-layout

**Reply by cnguu** [reply]
Nice！这正是我想要的
![image](https://user-images.githubusercontent.com/10758182/153130719-8c2e30a5-7c27-4e58-947c-b8518acc7807.png)

**Reply by zhaoquanjun** [reply]
你们的fixed列滑动的时候 不会错位么

**Reply by xzhdev** [reply]
设置table-layout：auto后固定列两个及以上会出现缝隙 ，有遇到嘛
#15298
